packages/components/src/components/link-tile/link-tile-utils.ts

```diff
@@ -1,25 +1,27 @@
-import { getTagName } from '../../utils';
-import type { BreakpointCustomizable } from '../../utils/breakpoint-customizable';
+import type { PropTypes } from '../../utils/validation/validateProps';
+import type { ButtonTile } from '../button-tile/button-tile';
+import type { LinkTile } from './link-tile';
+import {
+  AllowedTypes,
+  LINK_BUTTON_TILE_ALIGNS,
+  LINK_BUTTON_TILE_ASPECT_RATIOS,
+  LINK_BUTTON_TILE_SIZES,
+  LINK_BUTTON_TILE_WEIGHTS,
+  LinkTileAlign,
+  LinkTileAspectRatio,
+  LinkTileSize,
+  LinkTileWeight,
+} from '../../utils';
 
-export const LINK_TILE_WEIGHTS = ['regular', 'semibold'] as const;
-export type LinkTileWeight = typeof LINK_TILE_WEIGHTS[number];
-
-export const LINK_TILE_SIZES = ['default', 'inherit'] as const;
-export type LinkTileSize = typeof LINK_TILE_SIZES[number];
-
-export const LINK_TILE_ALIGNS = ['top', 'bottom'] as const;
-export type LinkTileAlign = typeof LINK_TILE_ALIGNS[number];
-
-export const LINK_TILE_ASPECT_RATIOS = ['1:1', '4:3', '3:4', '16:9', '9:16'] as const;
-export type LinkTileAspectRatio = typeof LINK_TILE_ASPECT_RATIOS[number];
-
-// does not take care of breakpoint customizable
-export const throwIfAlignTopAndNotCompact = (
-  host: HTMLElement,
-  align: LinkTileAlign,
-  compact: BreakpointCustomizable<boolean>
-): void => {
-  if (align === 'top' && (!compact || (typeof compact === 'string' && compact === 'false'))) {
-    throw new Error(`Usage of ${getTagName(host)} is not valid. Top alignment is only possible when compact is true.`);
-  }
+export const sharedTilePropTypes:
+  | Omit<PropTypes<typeof ButtonTile>, 'type' | 'disabled' | 'loading' | 'icon' | 'iconSource' | 'aria'>
```

Comment:
Would be great to have this without manual maintenance so that it automatically takes the common properties of both types. Something like  https://stackoverflow.com/questions/47375916/typescript-how-to-create-type-with-common-properties-of-two-types